model/test_results.go

```diff
@@ -295,42 +321,115 @@ func (t *TestResults) Download(ctx context.Context) ([]TestResult, error) {
 		t.ID = t.Info.ID()
 	}
 
-	var results testResultsDoc
-	catcher := grip.NewBasicCatcher()
-	bucket, err := t.GetBucket(ctx)
-	if err != nil {
-		return nil, err
-	}
-
 	switch t.Artifact.Version {
 	case 0:
+		catcher := grip.NewBasicCatcher()
+		bucket, err := t.GetBucket(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		var results []TestResult
 		iter := NewTestResultsIterator(bucket)
 		for iter.Next(ctx) {
-			results.Results = append(results.Results, iter.Item())
+			results = append(results, iter.Item())
 		}
-
 		catcher.Wrap(iter.Err(), "iterating test results")
 		catcher.Wrap(iter.Close(), "closing test results iterator")
+
+		return results, catcher.Resolve()
 	case 1:
-		r, err := bucket.Get(ctx, testResultsCollection)
+		bsonResults, err := t.downloadBSON(ctx)
 		if err != nil {
-			catcher.Wrap(err, "getting test results")
-			break
+			return nil, err
 		}
-
-		data, err := ioutil.ReadAll(r)
-		catcher.Add(r.Close())
-		if err != nil {
-			catcher.Wrap(err, "reading test results")
-			break
+		parquetResults, err := t.downloadParquet(ctx)
+		if err != nil && !pail.IsKeyNotFoundError(err) {
+			return nil, err
+		} else if err == nil && invalidParquetAlertCount < 5 {
+			if reflect.DeepEqual(parquetResults, bsonResults) {
+				grip.Warning(message.Fields{
+					"message":   "BSON and Parquet test results differ",
+					"task_id":   t.Info.TaskID,
+					"execution": t.Info.Execution,
+				})
+				invalidParquetAlertCount++
+			}
 		}
 
-		catcher.Wrap(bson.Unmarshal(data, &results), "unmarshalling test results")
+		return bsonResults, nil
 	default:
-		catcher.Errorf("unsupported test results artifact version '%d'", t.Artifact.Version)
+		return nil, errors.Errorf("unsupported test results artifact version '%d'", t.Artifact.Version)
+	}
+}
+
+// TODO (EVG-16140): Move this to Download (above) once we do the BSON to
```

Comment:
Does this mean that version 1 test results will only download from Presto eventually?

Reply:
Yeah, we have a backfill ticket to migrate all data within our TTL range to Presto so everything will be either in Presto or deleted.